src/vs/platform/terminal/common/terminalEnvironment.ts

```diff
@@ -4,26 +4,98 @@
 	let newPath = path;
 	if (newPath.includes('\\')) {
 		newPath = newPath.replace(/\\/g, '\\\\');
 	}
-	// Remove dangerous characters except single quotes, which we'll escape properly
-	const bannedChars = /[\`\$\|\&\>\~\#\!\^\*\;\<\"]/g;
+
+	// Shell-specific escaping
+	switch (shellType) {
+		case PosixShellType.Bash:
+		case PosixShellType.Sh:
+		case PosixShellType.Zsh:
+		case WindowsShellType.GitBash:
+			return escapeForPosixShell(newPath);
+		case PosixShellType.Fish:
+			return escapeForFishShell(newPath);
+		case GeneralShellType.PowerShell:
+			// PowerShell should be handled separately in preparePathForShell
+			// but if we get here, use PowerShell escaping
+			return escapeForPowerShell(newPath);
+		default:
+			// Default to POSIX shell escaping for unknown shells
+			return escapeForPosixShell(newPath);
+	}
+}
+
+/**
+ * Escape path for POSIX-compatible shells (bash, sh, zsh, git bash)
+ */
+function escapeForPosixShell(path: string): string {
+	let newPath = path;
+	
+	// Remove dangerous characters except single and double quotes, which we'll escape properly
+	const bannedChars = /[\`\$\|\&\>\~\#\!\^\*\;\<]/g;
 	newPath = newPath.replace(bannedChars, '');
 	
-	// Properly escape single quotes by ending the current single-quoted string,
-	// adding an escaped single quote, then starting a new single-quoted string
-	if (newPath.includes("'")) {
+	// If the path contains both single and double quotes, we need to use different escaping
+	if (newPath.includes("'") && newPath.includes('"')) {
+		// For paths with both quote types, escape using $'...' syntax
+		// This allows us to escape single quotes with \'
+		newPath = newPath.replace(/'/g, "\\'");
+		return `$'${newPath}'`;
+	} else if (newPath.includes("'")) {
+		// Only single quotes: use the standard POSIX method
 		newPath = newPath.replace(/'/g, "'\\''");
+		return `'${newPath}'`;
+	} else {
+		// No single quotes (may have double quotes): wrap in single quotes
+		return `'${newPath}'`;
+	}
+}
+
+/**
+ * Escape path for Fish shell
+ */
+function escapeForFishShell(path: string): string {
+	let newPath = path;
+	
+	// Remove dangerous characters except quotes
+	const bannedChars = /[\`\$\|\&\>\~\#\!\^\*\;\<]/g;
+	newPath = newPath.replace(bannedChars, '');
+	
+	// Fish shell supports escaping with backslashes for single quotes
+	// For double quotes, they're safe inside single quotes
+	if (newPath.includes("'")) {
+		newPath = newPath.replace(/'/g, "\\'");
+	}
+	
+	return `'${newPath}'`;
+}
+
+/**
+ * Escape path for PowerShell
+ */
+function escapeForPowerShell(path: string): string {
+	let newPath = path;
+	
+	// Remove dangerous characters except quotes
+	const bannedChars = /[\`\$\|\&\>\~\#\!\^\*\;\<]/g;
+	newPath = newPath.replace(bannedChars, '');
+	
+	// PowerShell uses double single quotes to escape single quotes
+	// Double quotes are safe inside single quotes
+	if (newPath.includes("'")) {
+		newPath = newPath.replace(/'/g, "''");
+		return `& '${newPath}'`;
```

Comment:
Dragging a path here doesn't necessarily mean to run it, so I don't think the & should not be here?